OpenRA.Mods.Common/Traits/Player/ProductionQueue.cs

```diff
@@ -470,9 +470,12 @@ public void EndProduction(ProductionItem item)
 			Queue.Remove(item);
 		}
 
-		protected void BeginProduction(ProductionItem item)
+		protected virtual void BeginProduction(ProductionItem item, bool queued)
```

Comment:
I wonder if it makes more sense to name the second argument `hasPriority` and switch the boolean check. I think that the name `queued` is a bit confusing.

Reply:
Looking at what's coming in #15021 (a variable named `queued` in the same method) I decided to do just that for more clarity.